### PR TITLE
Sync upstream and clarify examples usage

### PR DESCRIPTION
**Pull request by ten3roberts**
Brings sec-http3 up to speed with the final changes before merging in `h3` that occurred after merging.

The extra files as a result of renaming will be cleaned up

### CODE REVIEW

**Comment by ten3roberts** [comment]
The differences that this fork didn't catch as it was forked just before the final commits were minor and mostly touch ups.

**Comment by ten3roberts** [comment]
I refined the README inside examples and corrected the certificate paths.

Perhaps we should not Squash these commits as we would lose that difference, though that is just mainly preference.

I haven't used `Squash and Merge` much, as I've mostly used and do prefer to keep and refer back to specific commits and fixes.

**Comment by darioalessandro** [comment]
lgtm I enabled merge commits with linear history

**Review by darioalessandro — APPROVED** [review]
lgtm

**Comment by darioalessandro** [comment]
If you want to release this to crates.io you might want to bump the version.

File: sec-http3/src/config.rs [hunk 1]
```diff
@@ -2,31 +2,53 @@ use crate::proto::varint::VarInt;
...
 /// Configures the HTTP/3 connection
 #[derive(Debug, Clone, Copy)]
+#[non_exhaustive]
 pub struct Config {
     /// Just like in HTTP/2, HTTP/3 also uses the concept of "grease"
     /// to prevent potential interoperability issues in the future.
     /// In HTTP/3, the concept of grease is used to ensure that the protocol can evolve
     /// and accommodate future changes without breaking existing implementations.
-    pub send_grease: bool,
+    pub(crate) send_grease: bool,
```
Comment (ten3roberts) [inline sec-http3/src/config.rs:11]: My leave the user to configure.

Right now the `Config` struct as a whole is not accessible by name (sealed) so it does not make any difference right now.

**Comment by ten3roberts** [comment]
Mayhaps we should see if the custom webtransport session required by *salvo* would like to see an appearance in this crate.

This PR contains almost no api changes, and only small comment changes for links so it may not be worth it to publish a new version just right now, but we can come back to that.

**Comment by darioalessandro** [comment]
sounds good, smash that merge button whenever you are ready, I enabled both squash and merge and rebase and merge so that you can keep the commit history.

**Pull request merged**